lib/workflows/email/send_decom_flavor_email.py

```diff
@@ -0,0 +1,207 @@
+        .run(
+            cloud_account,
+            as_admin=True,
+            from_projects=from_projects if from_projects else None,
+            all_projects=not from_projects,
+        )
+        .append_from("PROJECT_QUERY", cloud_account, ProjectProperties.PROJECT_NAME)
+        .select(
+            ServerProperties.SERVER_ID,
+            ServerProperties.SERVER_NAME,
+            ServerProperties.ADDRESSES,
+        )
+    )
+
+    # run a user query from server query using then()
+    user_query = (
+        server_query.then("USER_QUERY", True)
+        .run(cloud_account)
+        .select(UserProperties.USER_NAME)
+        .group_by(UserProperties.USER_EMAIL)
+    )
+
+    return user_query
+
+
+def print_email_params(
+    email_addr: str, user_name: str, as_html: bool, flavor_table: str, decom_table: str
+):
+    """
+    Print email params instead of sending the email
+    :param email_addr: email address to send to
+    :param user_name: name of user in openstack
+    :param as_html: if the email is to be sent as html email
+    :param flavor_table: a table representing decommissioned flavors
+    :param decom_table: a table representing info found in openstack
+    about VMs running with decommissioned flavors
+    """
+    print(
+        f"Send Email To: {email_addr}\n"
+        f"email_templates decom-email: username {user_name}\n"
+        f"send as html: {as_html}\n"
+        f"flavor table: {flavor_table}\n"
+        f"decom table: {decom_table}\n"
+    )
+
+
+def build_email_params(
+    user_name: str, flavor_table: str, decom_table: str, **email_kwargs
+):
+    """
+    build email params dataclass which will be used to configure how to send the email
+    :param user_name: name of user in openstack
+    :param flavor_table: a table representing decommissioned flavors
+    :param decom_table: a table representing info found in openstack about VMs
+        running with decommissioned flavors
+    :param email_kwargs: a set of email kwargs to pass to EmailParams
+    """
+    body = EmailTemplateDetails(
+        template_name="decom_email",
+        template_params={
+            "username": user_name,
+            "affected_flavors_table": flavor_table,
+            "decom_table": decom_table,
+        },
+    )
+
+    footer = EmailTemplateDetails(template_name="footer", template_params={})
+
+    return EmailParams.from_dict(
+        {**{"email_templates": [body, footer]}, **email_kwargs}
+    )
+
+
+# pylint:disable=too-many-arguments
+# pylint:disable=too-many-locals
+def send_decom_flavor_email(
+    smtp_account: SMTPAccount,
+    cloud_account: Union[CloudDomains, str],
+    flavor_name_list: List[str],
+    from_projects: Optional[List[str]] = None,
+    all_projects: bool = False,
+    as_html: bool = False,
+    send_email: bool = False,
+    override_email: bool = False,
+    override_email_address: str = "cloud-support@stfc.ac.uk",
+    **email_params_kwargs,
```

Comment:
yea but this will feed directly from stackstorm. 
We discussed this earlier but we want to reduce actions/src to just this:

```
import workflows

def run(
    submodule_name: str,
    action_name: str,
    **kwargs
):
    """
    Dynamically dispatches to the function wanted
        :param submodule_name: submodule name which corresponds to module in workflow
        :param action_name: name of file/function which corresponds to function that will handle the action
        :param kwargs: All user-defined kwargs to pass to the function
     """
     submodule = getattr(workflows, submodule_name)

     action_module = getattr(action_name, submodule)
     action_func = getattr(action_name, action_module)

     return action_func(**kwargs)
```